assembly/buffer/index.ts

```diff
@@ -105,6 +108,53 @@ export class Buffer extends Uint8Array {
     store<u16>(this.dataStart + offset, bswap<u16>(value));
     return offset + 2;
   }
+
+  inspect(): string {
+    let byteLength = this.byteLength;
+    if (INSPECT_MAX_BYTES == 0 || byteLength == 0) return "<Buffer >";
+
+    // Calculate if an elipsis will be in the string
+    let elipsisEnd = byteLength > INSPECT_MAX_BYTES;
+    let maxBytes = elipsisEnd ? INSPECT_MAX_BYTES : byteLength;
+
+    // find the start of the buffer
+    let dataStart = this.dataStart;
+
+    // formula for calculating end string length (3 * bytes) + 8
+    // Example: Buffer.from([1, 2, 3, 4, 5]).inspect() == '<Buffer 01 02 03 04 05>'
+    let stringLength = 3 * maxBytes + 8;
+    if (elipsisEnd) stringLength += 3; // add 3 characters for elipsis
+
+    // create the result
+    let result = __alloc(stringLength << 1, idof<String>());
+
+    // copy the 16 "<Buffer " bytes
+    memory.copy(
+      result,
+      changetype<usize>(BUFFER_INSPECT_HEADER_START),
+      changetype<BLOCK>(changetype<usize>(BUFFER_INSPECT_HEADER_START) - BLOCK_OVERHEAD).rtSize,
+    );
+
+    // Start writing at index 8
+    let writeOffset: usize = result + 16;
+    for (let i = 0; i < maxBytes; i++, writeOffset += 6) {
+      let byte = <u32>load<u8>(dataStart + <usize>i);
+
+      store<u32>(writeOffset, Buffer.HEX.charsFromByte(byte));
+      if (i == (maxBytes - 1)) {
+        if (elipsisEnd) {
+          // make this a single 64 bit store
+          store<u64>(writeOffset, <u64>17451646127570990, 4); // "...>"
```

Comment:
Could you use `0x3E_002E_002E_002E` instead `17451646127570990`?

Reply:
```
> (17451646127570990).toString(16)
'3e002e002e002e'
```

Yes. I could use this.